app/src/main/java/org/thoughtcrime/securesms/conversation/v2/input_bar/InputBarRecordingView.kt

```diff
@@ -132,7 +132,12 @@ class InputBarRecordingView : RelativeLayout {
     private fun updateTimer() {
         val duration = (Date().time - startTimestamp) / 1000L
         binding.recordingViewDurationTextView.text = DateUtils.formatElapsedTime(duration)
-        snHandler.postDelayed({ updateTimer() }, 500)
+
+        if (isAttachedToWindow) {
+            // Should only update the timer if the view is still attached to the window.
+            // Otherwise, the timer will keep running even after the view is detached.
+            snHandler.postDelayed({ updateTimer() }, 500)
```

Comment:
Yeah I did check it still works. Originally I had the same thought as I didn't know whether putting a view into background will count as a detach but it seems like the window still exists in the background. 

Just for a fun reference, this is an answer I got from chatgpt:
> No, putting an Android activity into the background does not destroy its window. The activity's window is only destroyed when the activity is finished (i.e., its finish() method is called) or the system needs to reclaim memory and decides to kill the activity's process.  When an activity goes into the background, it is in the stopped state but its state is preserved. This means that the system maintains all view hierarchy and data associated with the activity. The activity's window is hidden, but all of its state such as member variables is preserved.  However, it's important to note that when the system is low on memory, it may decide to kill background activities to reclaim resources. In this case, the activity's window and state would be destroyed. The system would later recreate the activity if it becomes visible again, calling onCreate() with a non-null Bundle that has the saved state.

Reply:
ok, still a little brittle to stop on remove, and not start again if a dev was

There's another issue:

`ConActivityV2#showVoiceMessageUI` calls `InputBarRecordingView#show` calls `updateTimer()`

So every time we start a new voice recording we add another one of these async-recursive calls.

Maybe we can address that too?

Reply:
Hmm ok I can try to address the "what if the dev wants to re-use the view but keep animation running" scenario, I was been lazy.

I'm not sure about your second question, `updateTimer` doesn't seem to "add" anything?

Reply:
`updateTimer` posts a delayed call to `updateTimer`

Reply:
Ah I see what you mean: a duplicated call into `show` will have started another timer, which will do nothing but increase the frequency of timer update?

Reply:
Yeah, every time you record a message, we'll create a new "async-recursive" loop.